components/AdditionalDatasheetEditor.tsx

```diff
@@ -193,54 +212,57 @@ export function AdditionalDatasheetEditor() {
     [updateMeasureDataPoint, setNotification, selectedInstanceId]
   );
 
-  const COLUMNS: GridColDef[] = [
-    {
-      headerName: '',
-      field: 'question',
-      flex: 3,
-      renderCell: (params) => (
-        <Typography
-          variant="body2"
-          style={{ whiteSpace: 'normal', lineHeight: 1.5 }}
-          sx={{
-            pr: 5,
-          }}
-        >
-          {params.value}
-        </Typography>
-      ),
-    },
-    {
-      headerName: 'Baseline',
-      field: 'baselineYear',
-      flex: 1,
-      renderCell: (params) => (
-        <Typography variant="body2">{params.value}</Typography>
-      ),
-      renderHeader: () => (
-        <Box sx={{ textAlign: 'center', lineHeight: 1.5 }}>
-          <Typography variant="body" component="div">
-            {baselineYear}
-          </Typography>
-          <Typography variant="caption" component="div">
-            Baseline
+  const COLUMNS: GridColDef[] = useMemo(
+    () => [
+      {
+        headerName: '',
+        field: 'question',
+        flex: 3,
+        renderCell: (params: GridRenderCellParams) => (
+          <Typography
+            variant="body2"
+            style={{ whiteSpace: 'normal', lineHeight: 1.5 }}
+            sx={{ pr: 5 }}
+          >
+            {params.value}
           </Typography>
-        </Box>
-      ),
-    },
-    { headerName: 'Unit', field: 'unit', flex: 1 },
-    ...additionalYears.map((year) => ({
-      headerName: year.toString(),
-      field: year.toString(),
-      editable: true,
-      flex: 1,
-      renderCell: (params) => (
-        <CustomEditComponent {...params} sx={{ mx: 0, my: 0 }} />
-      ),
-    })),
-  ];
+        ),
+      },
+      {
+        headerName: 'Baseline',
+        field: 'baselineYear',
+        flex: 1,
+        renderCell: (params: GridRenderCellParams) => (
+          <Typography variant="body2">{params.value}</Typography>
+        ),
+        renderHeader: () => (
+          <Box sx={{ textAlign: 'center', lineHeight: 1.5 }}>
+            <Typography variant="body2" component="div">
+              {baselineYear}
+            </Typography>
+            <Typography variant="caption" component="div">
+              Baseline
+            </Typography>
+          </Box>
+        ),
+      },
+      { headerName: 'Unit', field: 'unit', flex: 1 },
+      ...additionalYears.map((year) => ({
+        headerName: year.toString(),
+        field: year.toString(),
+        editable: true,
+        flex: 1,
+        renderCell: (params: GridRenderCellParams) => (
+          <CustomEditComponent {...params} sx={{ mx: 0, my: 0 }} />
+        ),
+      })),
+    ],
+    [baselineYear, additionalYears]
+  );
+
+  const isLoading = queryLoading || mutationLoading;
 
-  if (loading) {
+  if (isLoading) {
```

Comment:
This will render the skeleton loader and replace the datagrid whenever the mutation is loading, instead we should only show the skeleton on `queryLoading`

 ```suggestion
  if (queryLoading) {
```